scipy/optimize/_differentialevolution.py

```diff
@@ -578,13 +595,20 @@ def __init__(self, func, bounds, args=(),
         # least 5 long
         self.num_population_members = max(5, popsize * self.parameter_count)
 
+        if isinstance(init, str) and (init == 'sobol'):
+            # must be Ns=2**m for Sobol'
+            n_s = int(2 ** np.ceil(np.log2(self.num_population_members)))
```

Comment:
This has the effect of overriding the docstring:

> The population has popsize * len(x) individuals`.

 Perhaps a note should be added to that effect somewhere.

Reply:
I added a note in the description of `popsize` and changed the wording on the `init` part.